mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -175,6 +178,9 @@ public void setInMailboxes(MessageId messageId, List<MailboxId> mailboxIds, Mail
             List<MailboxId> mailboxesToRemove = ImmutableList.copyOf(Sets.difference(currentMailboxes, targetMailboxes));
             SetView<MailboxId> mailboxesToAdd = Sets.difference(targetMailboxes, currentMailboxes);
 
+            assertRightsOnMailboxes(mailboxesToAdd, mailboxSession, MailboxACL.Right.Insert);
```

Comment:
why these two lines are not a single call ?

Reply:
Rights are different, mailboxes are different?

I really don't know what you have in mind!

Reply:
Oh, right, I misread